### ISSUE TITLE
[NVPTX] Unhandled source type UNREACHABLE executed at NVPTXISelDAGToDAG.cpp:3808!

### ISSUE THREAD

**Issue by d0k** [llvm/llvm-project#98033]
```
target datalayout = "e-i64:64-i128:128-v16:16-v32:32-n16:32:64"
target triple = "nvptx64"

define void @loop_slice_fusion_5(ptr noalias readonly %0) {
._crit_edge:
  %1 = load i1, ptr %0, align 1
  %2 = zext i1 %1 to i32
  %3 = add i32 %2, 1
  %4 = and i32 %3, 1
  store i32 %4, ptr null, align 2
  ret void
}

!nvvm.annotations = !{!0, !1}

!0 = !{ptr @loop_slice_fusion_5, !"kernel", i32 1}
!1 = !{ptr @loop_slice_fusion_5, !"reqntidx", i32 128}
```

```
$ llc < t.ll -mcpu=sm_80

Unhandled source type
UNREACHABLE executed at llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp:3808!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: llc -mcpu=sm_80 -debug
1.      Running pass 'Function Pass Manager' on module '<stdin>'.
2.      Running pass 'NVPTX DAG->DAG Pattern Instruction Selection' on function '@loop_slice_fusion_5'
 #0 0x00005606095885d4 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (llc+0x69e15d4)
 #1 0x000056060958617e llvm::sys::RunSignalHandlers() (llc+0x69df17e)
 #2 0x0000560609588978 SignalHandler(int) Signals.cpp:0:0
 #3 0x00007f1a26e5a510 (/lib/x86_64-linux-gnu/libc.so.6+0x3c510)
 #4 0x00007f1a26ea816c __pthread_kill_implementation ./nptl/pthread_kill.c:44:76
 #5 0x00007f1a26e5a472 raise ./signal/../sysdeps/posix/raise.c:27:6
 #6 0x00007f1a26e444b2 abort ./stdlib/abort.c:81:7
 #7 0x00005606094f2e30 llvm::install_out_of_memory_new_handler() (llc+0x694be30)
 #8 0x00005606077c221d llvm::NVPTXDAGToDAGISel::GetConvertOpcode(llvm::MVT, llvm::MVT, llvm::LoadSDNode*) NVPTXISelDAGToDAG.cpp:0:0
 #9 0x00005606077bcd05 llvm::NVPTXDAGToDAGISel::tryLDGLDU(llvm::SDNode*) NVPTXISelDAGToDAG.cpp:0:0
#10 0x00005606077b892f llvm::NVPTXDAGToDAGISel::tryLoad(llvm::SDNode*) NVPTXISelDAGToDAG.cpp:0:0
#11 0x00005606077b8747 llvm::NVPTXDAGToDAGISel::Select(llvm::SDNode*) NVPTXISelDAGToDAG.cpp:0:0
#12 0x000056060938eb9f llvm::SelectionDAGISel::DoInstructionSelection() (llc+0x67e7b9f)
#13 0x000056060938dba9 llvm::SelectionDAGISel::CodeGenAndEmitDAG() (llc+0x67e6ba9)
#14 0x000056060938b50c llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (llc+0x67e450c)
#15 0x00005606093887aa llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (llc+0x67e17aa)
#16 0x0000560609386146 llvm::SelectionDAGISelLegacy::runOnMachineFunction(llvm::MachineFunction&) (llc+0x67df146)
#17 0x000056060868bec2 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (llc+0x5ae4ec2)
#18 0x0000560608b9e5bc llvm::FPPassManager::runOnFunction(llvm::Function&) (llc+0x5ff75bc)
#19 0x0000560608ba6c22 llvm::FPPassManager::runOnModule(llvm::Module&) (llc+0x5fffc22)
#20 0x0000560608b9f137 llvm::legacy::PassManagerImpl::run(llvm::Module&) (llc+0x5ff8137)
#21 0x00005606069dccf3 main (llc+0x3e35cf3)
#22 0x00007f1a26e456ca __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:74:3
#23 0x00007f1a26e45785 call_init ./csu/../csu/libc-start.c:128:20
#24 0x00007f1a26e45785 __libc_start_main ./csu/../csu/libc-start.c:347:5
#25 0x00005606069d6c21 _start (llc+0x3e2fc21)
```

**Comment by mjulian31** [comment]
@d0k - this issue should now be fixed by my commit #99392. Please check if you are still seeing the bug.

**Comment by d0k** [comment]
Issue is gone, thanks.
